scripts/Test-LCOW-UVM.ps1

```diff
@@ -0,0 +1,149 @@
+
+    [string]
+    $Timeout = '10m',
+
+    [alias('tv')]
+    [switch]
+    $TestVerbose,
+
+    [string]
+    $Run = '',
+
+    [string]
+    $CodePath = '.',
+
+    [string]
+    $OutDirectory = '.\test\results',
+
+    # uvm parameters
+
+    [string]
+    $BootFilesPath = 'C:\ContainerPlat\LinuxBootFiles',
+
+    [ValidateSet('vhd', 'initrd')]
+    [string]
+    $BootFSType = 'vhd',
+
+    [switch]
+    $DisableTimeSync,
+
+    # gcs test/container options
+
+    [string]
+    $ContainerRootFSMount = '/run/rootfs',
+
+    [string]
+    $ContainerRootFSPath = (Join-Path $BootFilesPath 'rootfs.vhd'),
+
+    [string]
+    $GCSTestMount = '/run/bin',
+
+    [string]
+    $GCSTestPath = '.\bin\test\gcstest',
+
+    [switch]
+    $MountGCSTest,
+
+    [string]
+    $Feature = ''
+)
+
+Import-Module ( Join-Path $PSScriptRoot Testing.psm1 ) -Force
+
+$CodePath = Resolve-Path $CodePath
+$OutDirectory = Resolve-Path $OutDirectory
+$BootFilesPath = Resolve-Path $BootFilesPath
+$ContainerRootFSPath = Resolve-Path $ContainerRootFSPath
+$GCSTestPath = Resolve-Path $GCSTestPath
+
+$shell = ( $Action -eq 'Shell' )
+
+if ( $shell ) {
+    $cmd = 'ash'
+} else {
+    $date = Get-Date
+    $waitfiles = "$ContainerRootFSMount"
+    $gcspath = 'gcstest'
+    if ( $MountGCSTest ) {
+        $waitfiles += ",$GCSTestMount"
+        $gcspath = "$GCSTestMount/gcstest"
+    }
+
+    $pre = "wait-paths -p $waitfiles -t 5 ; " + `
+        'echo nproc: `$(nproc) ; ' + `
+        'echo kernel: `$(uname -a) ; ' + `
+        'echo gcs.commit: `$(cat /info/gcs.commit 2>/dev/null) ; ' + `
+        'echo gcs.branch: `$(cat /info/gcs.branch 2>/dev/null) ; ' + `
+        'echo tar.date:   `$(cat /info/tar.date 2>/dev/null) ; ' + `
+        'echo image.name: `$(cat /info/image.name 2>/dev/null) ; ' + `
+        'echo build.date: `$(cat /info/build.date 2>/dev/null) ; '
+
+    $testcmd, $out = New-TestCommand `
+        -Action $Action `
+        -Path $gcspath `
+        -Name gcstest `
+        -OutDirectory $OutDirectory `
+        -Date $date `
+        -Note $Note `
+        -TestVerbose:$TestVerbose `
+        -Count $Count `
+        -BenchTime $BenchTime `
+        -Timeout $Timeout `
+        -Run $Run `
+        -Feature $Feature `
+        -Verbose:$Verbose
+
+    $testcmd += " `'-rootfs-path=$ContainerRootFSMount`' "
+    $cmd = $pre + $testcmd
+}
+
+$boot = '.\bin\tools\uvmboot.exe -gcs lcow ' + `
+    '-fwd-stdout -fwd-stderr -output-handling stdout ' + `
+    "-boot-files-path $BootFilesPath " + `
+    "-root-fs-type $BootFSType " + `
+    '-kernel-file vmlinux ' + `
+    "-security-policy=`"`" " + `
+    "-mount `"$ContainerRootFSPath,$ContainerRootFSMount`" "
+
+if ( $MountGCSTest ) {
+    $boot += "-share `"$GCSTestPath,$GCSTestMount`" "
+}
+
+if ( $DisableTimeSync ) {
+    $boot += ' -disable-time-sync '
+}
+
+if ( $shell ) {
+    $boot += ' -t '
+}
+
+$boot += " -exec `"$cmd`" "
+
+Invoke-TestCommand `
+    -TestCmd $boot `
+    -TestCmdPreamble $testcmd `
```

Comment:
Doesn't $boot include $cmd which has $testcmd?

Reply:
yes, but `-TestCmdPreamble` just writes out the string argument before execing the tests.
its a bit convoluted, but this way, before the test output runs, we get headers with the test info, including the command that was run, so we can just save the output on its own, if need be, and still have all the data we need:
```
date: 20220412T1701329409
note: 46-117
test.command: .\bin\test\functional.exe -feature="LCOW"  "-test.shuffle=on" "-test.run=^#" "-test.bench=."  "-test.benchmem" "-test.benchtime=5s" "-test.count=5" 
repo.commit: 5631c613afeabbac116133f06a4b2ca17fe83d8e
-test.shuffle 1649797293380880100
goos: windows
goarch: amd64
pkg: github.com/Microsoft/hcsshim/test/functional
cpu: Intel(R) Xeon(R) W-2235 CPU @ 3.80GHz
BenchmarkLCOWStart-12                            	       7	 806345986 ns/op	   16400 B/op	     157 allocs/op
# ...
```

If we printed `$boot`, it would have all the boot and mount and a bunch of other redundant info